### PR TITLE
Add Set Pattern block

### PR DESCRIPTION
**Pull request by kozzi**
This change adds the Set Pattern block to artist.

pattern-artist2

Most of the work is actually already in place and ready to go. A lot of this change simply wired it all back together so that it can be used in artist.

Note: There is a lot of custom work done for the Frozen skins that make them not work well with patterns other than their own lines. During testing, I found that when Anna/Elsa drew other patterns, they all look distorted or is weirdly offset. So for the Anna and Elsa skin, I don't allow patterns other than their own in the Set Patterns block dropdown.
### Testing
- Tested this with the artist, elsa, and anna skins.
- Tested the Frozen tutorial for good measure.
- Ran `grunt mochaTest` per [the testing doc](https://github.com/code-dot-org/code-dot-org/blob/staging/TESTING.md#mocha-tests) 
- This is my first time make changes in this area; are there other things I should test?
### Todo
- We probably need to replace all the line art assets

### CODE REVIEW

File: apps/src/turtle/skins.js [hunk 1]
```diff
@@ -42,18 +42,29 @@ exports.load = function (assetUrl, id) {
...
+  skin.lineStylePatternOptions = [[skin.patternDefault, 'DEFAULT']];
+
+  // Then selectively include other line patterns depending on the skin.
+  // Anna and Elsa probably shouldn't draw other line patterns other than their own.
+  // When they try to draw other patterns, it looks very distorted.
+  // This is likely because there was lot hand-tweaking done previously to make those
+  // line patterns look good, and we don't want to touch that.
+
+  // With Anna and Elsa, we're only including their own patterns in the dropdown.
+  if (skin.id == "anna") {
```
Comment (breville) [inline apps/src/turtle/skins.js]: how about using the CONFIGS array above instead of adding new branching logic here?

**Comment by breville** [comment]
Anna & Elsa patterns are rendered based on the fact that we draw their lines step by step rather than all at once.  Hopefully one day we can add that feature to regular artist too.

btw, those two skins should only ever appear as part of the frozen tutorial, and not be selectable anywhere else.

As discussed by email, we need to create our own pattern images.

One suggestion, otherwise lgtm.

File: apps/src/turtle/skins.js (hunk 1 above)
Comment (kozzi) [inline apps/src/turtle/skins.js]: Good point. Fixed.

**Comment by kozzi** [comment]
Follow-up note: Tanya reached out to Barry and confirmed that we can use existing assets, so going ahead and merging to staging. I updated the rainbow pattern to be more 'rainbow'-y, but kept the rest of it as-is.

**Pull request merged**